test1/task1/main.c

```diff
@@ -0,0 +1,124 @@
+﻿#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <limits.h>
+
+#define NUMBER_OF_BITS sizeof(unsigned int) * 8
+
+enum errorCodes {
+    testFailed = -1,
+    success = 0,
+    badAllocation = 1
+};
+
+static int compare(const bool* const bits1, const bool* const bits2)
+{
+    for (size_t i = 0; i < NUMBER_OF_BITS; ++i)
+    {
+        if (bits1[i] != bits2[i])
+        {
+            return bits1[i] ? -1 : 1;
+        }
+    }
+    return 0;
+}
+
+static bool* convertToBinary(const unsigned int number)
+{
+    bool* bits = (bool*)malloc(NUMBER_OF_BITS);
+    if (bits == NULL)
+    {
+        return NULL;
+    }
+    unsigned int bit = 1;
+    for (size_t i = 0; i < NUMBER_OF_BITS; ++i)
+    {
+        bits[NUMBER_OF_BITS - i - 1] = number & bit ? true : false;
```

Comment:
```suggestion
        bits[NUMBER_OF_BITS - i - 1] = number & bit != 0;
```
Это то же самое, только короче и не так странно (всегда, когда считается какое-то логическое условие и возвращается true/false константами, это выглядит диковато, потому что это же результат вычисления логического условия).